tests/check/check_test.go

```diff
@@ -126,3 +132,64 @@ func runTests(t *testing.T, client pb.OpenFGAServiceClient, tests checkTests) {
 		})
 	}
 }
+
+func testBadAuthModelID(t *testing.T, engine string) {
+
+	cfg := run.MustDefaultConfigWithRandomPorts()
+	cfg.Log.Level = "none"
+	cfg.Datastore.Engine = engine
+
+	cancel := tests.StartServer(t, cfg)
+	defer cancel()
+
+	conn := tests.Connect(t, cfg.GRPC.Addr)
+	defer conn.Close()
+	client := pb.NewOpenFGAServiceClient(conn)
+
+	ctx := context.Background()
+	resp, err := client.CreateStore(ctx, &pb.CreateStoreRequest{Name: "bad auth id"})
+	require.NoError(t, err)
+
+	storeID := resp.GetId()
+	_, err = client.WriteAuthorizationModel(ctx, &pb.WriteAuthorizationModelRequest{
+		StoreId:       storeID,
+		SchemaVersion: typesystem.SchemaVersion1_1,
+		TypeDefinitions: []*pb.TypeDefinition{
+			{
+				Type:      "user",
+				Relations: map[string]*pb.Userset{},
+			},
+			{
+				Type: "doc",
+				Relations: map[string]*pb.Userset{
+					"viewer":   typesystem.This(),
+					"can_view": typesystem.ComputedUserset("viewer"),
+				},
+				Metadata: &pb.Metadata{
+					Relations: map[string]*pb.RelationMetadata{
+						"viewer": {
+							DirectlyRelatedUserTypes: []*pb.RelationReference{
+								{
+									Type: "user",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+	_, err = client.Check(ctx, &pb.CheckRequest{
+		StoreId:              storeID,
+		TupleKey:             tuple.NewTupleKey("doc:x", "viewer", "user:y"),
+		AuthorizationModelId: "01GS89AJC3R3PFQ9BNY5ZF6Q97",
+	})
+
+	require.Error(t, err)
+	e, ok := status.FromError(err)
+	require.True(t, ok)
+	require.Equal(t, int(pb.ErrorCode_authorization_model_not_found), int(e.Code()))
+
+	cancel()
```

Comment:
Redundant. We have `defer cancel()` above.

Reply:
removed as suggested